controllers/btpoperator_controller.go

```diff
@@ -1131,3 +1201,480 @@ func (r *BtpOperatorReconciler) watchConfigPredicates() predicate.Funcs {
+	}
+	return true, nil
+}
+
+func (r *BtpOperatorReconciler) doFullyCertificatesRegenerate(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured) error {
+	logger := log.FromContext(ctx)
+	logger.Info("full regeneration started")
+
+	ca, caPk, err := r.generateSelfSignedCertAndAddToApplyList(ctx, resourcesToApply)
+	if err != nil {
+		return fmt.Errorf("error while generating self signed cert in full regeneration proccess. %w", err)
+	}
+
+	err = r.generateSignedCertAndAddToApplyList(ctx, resourcesToApply, ca, caPk)
+	if err != nil {
+		return fmt.Errorf("error while generating signed cert in full regeneration proccess. %w", err)
+	}
+
+	if err := r.reconcileWebhooks(ctx, resourcesToApply, ca); err != nil {
+		return fmt.Errorf("error while reconciling webhooks. %w", err)
+	}
+
+	logger.Info("full regeneration success")
+	return nil
+}
+
+func (r *BtpOperatorReconciler) doPartialCertificatesRegenerate(ctx context.Context, resourceToApply *[]*unstructured.Unstructured) error {
+	logger := log.FromContext(ctx)
+	logger.Info("partial regeneration started")
+
+	err := r.generateSignedCertAndAddToApplyList(ctx, resourceToApply, nil, nil)
+	if err != nil {
+		return fmt.Errorf("error while generating signed cert in partial regeneration proccess. %w", err)
+	}
+	logger.Info("partial regeneration success")
+	return nil
+}
+
+func (r *BtpOperatorReconciler) generateSelfSignedCertAndAddToApplyList(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured) ([]byte, []byte, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("generation of self signed cert started")
+
+	ca, caPk, err := certs.GenerateSelfSignedCertificate(time.Now().Add(CaCertificateExpiration))
+	if err != nil {
+		return []byte{}, nil, fmt.Errorf("while generating self signed cert: %w", err)
+	}
+
+	err = r.appendCertificationDataToUnstructured(resourcesToApply, CaSecret, ca, caPk, CaSecretDataPrefix)
+	if err != nil {
+		return []byte{}, nil, fmt.Errorf("while adding newly generated self signed cert to resoruces to apply: %w", err)
+	}
+
+	logger.Info("generation of self signed cert ok")
+	return ca, caPk, nil
+}
+
+func (r *BtpOperatorReconciler) generateSignedCertAndAddToApplyList(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured, CA, CAPrivateKey []byte) error {
+	logger := log.FromContext(ctx)
+	logger.Info("generation of signed webhookCertificate started")
+
+	webhookCertificate, webhookPrivateKey, err := r.generateSignedCert(ctx, time.Now().Add(WebhookCertificateExpiration), CA, CAPrivateKey)
+	if err != nil {
+		return fmt.Errorf("while generating signed webhookCertificate: %w", err)
+	}
+	err = r.appendCertificationDataToUnstructured(resourcesToApply, WebhookSecret, webhookCertificate, webhookPrivateKey, WebhookSecretDataPrefix)
+	if err != nil {
+		return fmt.Errorf("while adding newly generated signed webhookCertificate to resoruces to apply: %w", err)
+	}
+	logger.Info("generation of signed webhookCertificate ok")
```

Comment:
You introduced kind of convention in logging.
<something> started
<something> ok

but sometimes inconsistent form (verb + noun):
<something> started
<something> success

Please choose one.